sdk/src/transaction.rs

```diff
@@ -255,6 +255,61 @@ impl Transaction {
         }
     }
 
+    /// Check keys and keypair lengths, then sign this transaction, using an assortment of
+    /// KeypairUtil trait objects
+    pub fn sign_dynamic_signers(
+        &mut self,
+        keypairs: &[&dyn KeypairUtil],
+        recent_blockhash: Hash,
+    ) -> result::Result<(), Box<dyn error::Error>> {
```

Comment:
Why `Box<dyn error::Error>` and not something specific to the actual errors that are possible?

Reply:
Are you envisioning something like:
```
enum SigningError {
  PresignerError,
  RemoteWalletError,
  ..
}
```
Short and sweet for now, but forces any new signers that impl KeypairUtil to add a variation to the enum.
I suppose we could have a single `SigningError(String)` that wraps any signer error it receives, but is that any better than the Box?

Reply:
`SigningError(String)` would definitely be worse.  I think you should create that `enum` but I don't think it'll need many escape hatches.  Mayyyybe a CustomError, like we have in InstructionError.

Reply:
Just realized a problem with that enum. It would be best to bubble up the actual errors from signers (particularly RemoteKeypair which can throw a variety of different kinds of errors). Something like this would be nice with thiserror:
```
#[derive(Error, Debug)]
pub enum SigningError {
    #[error("remote wallet error")]
    RemoteWallet(#[from] remote_wallet::RemoteWalletError),
}
```
But to do that in transaction.rs, sdk would need to depend on the remote-wallet crate, which it can't, because remote-wallet depends on sdk. So where does SigningError live?
An alternative is a generic `SigningError::RemoteWallet`, but losing that error data is worse than Box<Error>, imho